lib/src/object_store.rs

```diff
@@ -30,15 +44,32 @@ impl ObjectStores {
 
     pub fn lookup(
         &self,
-        obj_store_key: &ObjectStoreKey,
-        obj_key: &ObjectKey,
-    ) -> Result<Vec<u8>, ObjectStoreError> {
+        obj_store_key: ObjectStoreKey,
+        obj_key: ObjectKey,
+    ) -> Result<ObjectValue, KvStoreError> {
+        let mut res = Err(KvStoreError::InternalError);
```

Comment:
This is due to the non-functional nature of the `and_modify` on L54. Both the logic branches set the response value, so you'll still get the `NotFound` when appropriate.

You've made me realize this should actually be an `Uninitialized` though. Here's the full new block:

```
        let mut res = Err(KvStoreError::Uninitialized);

        self.stores
            .write()
            .map_err(|_| KvStoreError::InternalError)?
            .entry(obj_store_key)
            .and_modify(|store| match store.get(&obj_key) {
                Some(val) => {
                    res = Ok(val.clone());
                    // manages ttl
                    if let Some(exp) = val.expiration {
                        if SystemTime::now() >= exp {
                            store.remove(&obj_key);
                            res = Err(KvStoreError::NotFound);
                        }
                    }
                }
                None => {
                    res = Err(KvStoreError::NotFound);
                }
            });

        res
```

so Uninitialized to catch the failure to implement, otherwise branch to Ok(res) or NotFound.